### PR TITLE
Added Automation integration demo

### PR DESCRIPTION
**Pull request by KuphJr**

### CODE REVIEW

File: contracts/dev/functions/FunctionsClient.sol
Comment (KuphJr) [inline contracts/dev/functions/FunctionsClient.sol:14]: These should be `internal`, not `private` as it allows a `FunctionsConsumer` contract to use these values directly.  This is necessary for `AutomatedFunctionsConsumer` to avoid running the expensive `encodeCBOR` for every request triggered by Automation.

Comment (KuphJr) [reply contracts/dev/functions/FunctionsClient.sol:14]: See the `AutomatedFunctionsConsumer` contract or `deployAutoClient.js` for details.

File: README.md
Comment (bolekk) [inline README.md:212]: Could this be a single command that reads either response or error?

File: contracts/AutomatedFunctionsConsumer.sol
Comment (bolekk) [inline contracts/AutomatedFunctionsConsumer.sol:140]: remove

File: contracts/FunctionsConsumer.sol
Comment (bolekk) [inline contracts/FunctionsConsumer.sol:90]: Let's not add tooling-specific comments to generic contracts.

File: contracts/dev/functions/FunctionsClient.sol
Comment (bolekk) [reply contracts/dev/functions/FunctionsClient.sol:14]: Please remember to port contract changes to core repo afterwards (or even better before merging this PR).

File: tasks/Functions-client/deployAutoClient.js
Comment (bolekk) [inline tasks/Functions-client/deployAutoClient.js:34]: I think we can skip this part.

File: tasks/Functions-client/deployAutoClient.js
Comment (bolekk) [inline tasks/Functions-client/deployAutoClient.js:106]: Reuse code from add.js

File: tasks/Functions-client/deployAutoClient.js
Comment (bolekk) [inline tasks/Functions-client/deployAutoClient.js:156]: Reuse code

File: tasks/Functions-client/performManualUpkeep.js
Comment (bolekk) [inline tasks/Functions-client/performManualUpkeep.js:21]: remove

**Review by bolekk — CHANGES_REQUESTED** [review]

File: README.md
Comment (KuphJr) [reply README.md:212]: Maybe.  We could make a separate PR.

File: contracts/dev/functions/FunctionsClient.sol
Comment (justinkaseman) [reply contracts/dev/functions/FunctionsClient.sol:14]: Let's do core repo first, opened here: https://github.com/smartcontractkit/chainlink/pull/8429

**Comment by justinkaseman** [comment]
LGTM, only thing I notice is that depending on if this goes in first or Upgrades go in first we will need a few tweaks to:
- Read from proxy address, network-config changes
- Use exact path for contract factories because now there is a name clash from the migration (also just good to do in general)

I'll let @bolekk give a re-review before we merge

File: tasks/Functions-client/deployAutoClient.js
Comment (justinkaseman) [inline tasks/Functions-client/deployAutoClient.js:54]:
```suggestion
    const OracleFactory = await ethers.getContractFactory("FunctionsOracleProxy")
```

File: tasks/Functions-client/deployAutoClient.js
Comment (justinkaseman) [inline tasks/Functions-client/deployAutoClient.js:55]:
```suggestion
    const oracle = await OracleFactory.attach(networkConfig[network.name]["functionsOracleProxy"])
```

File: tasks/Functions-billing/add.js
Comment (justinkaseman) [inline tasks/Functions-billing/add.js:19]:
```suggestion
  const registry = await RegistryFactory.attach(networkConfig[network.name]["functionsOracleRegistryProxy"])
```

File: tasks/Functions-client/deployAutoClient.js
Comment (justinkaseman) [inline tasks/Functions-client/deployAutoClient.js:36]:
```suggestion
    const oracleAddress = networkConfig[network.name]["functionsOracleProxy"]
```

**Review by justinkaseman — CHANGES_REQUESTED** [review]
Changes from the Proxy PR that went in

File: README.md
Comment (justinkaseman) [inline README.md:205]:
```suggestion
URLs which host secrets must be available every time a request is executed by DON nodes. For optimal security, it is recommended to expire the URLs when the off-chain secrets are no longer in use.
```

**Review by justinkaseman — APPROVED** [review]
LGTM, good stuff

**Review by bolekk — APPROVED** [review]

**Pull request merged**
